controllers/operator/common_controller.go

```diff
@@ -97,7 +98,31 @@ func NewReconcileCommonController(ctx context.Context, client client.Client) *Re
 	}
 }
 
-func ensureRoles(roles []mdbv1.MongoDbRole, conn om.Connection, log *zap.SugaredLogger) workflow.Status {
+// ensureRoles will first check if both roles and roleRefs are populated. If both are, it will return an error, which is inline with the webhook validation rules.
+// Otherwise, if roles is populated, then it will extract the list of roles and check if they are already set in Ops Manager. If they are not, it will update the roles in Ops Manager.
+// If roleRefs is populated, it will extract the list of roles from the referenced resources and check if they are already set in Ops Manager. If they are not, it will update the roles in Ops Manager.
+func (r *ReconcileCommonController) ensureRoles(ctx context.Context, db mdbv1.DbCommonSpec, watchClusterMongoDBRoles bool, conn om.Connection, mongodbResourceNsName types.NamespacedName, log *zap.SugaredLogger) workflow.Status {
+	localRoles := db.GetSecurity().Roles
+	roleRefs := db.GetSecurity().RoleRefs
+
+	if len(localRoles) > 0 && len(roleRefs) > 0 {
+		return workflow.Failed(xerrors.Errorf("At most one one of roles or roleRefs can be non-empty."))
+	}
+
+	var roles []mdbv1.MongoDBRole
+	if len(roleRefs) > 0 {
+		if !watchClusterMongoDBRoles {
+			return workflow.Failed(xerrors.Errorf("RoleRefs are not supported when ClusterMongoDBRoles are disabled. Please enable ClusterMongoDBRoles in the operator configuration."))
```

Comment:
We should tell in the message exactly what parameter to set or point to the documentation. In this case it will be helm's values.yaml parameter to enable cluster roles. 

But for someone who installs the operator manually using yaml and kubectl it will be confusing. As what is actually required here to pass this check is to add watch-resource parameter. 

What's worse - even without watching the resource, references might work because the only thing the operator needs is to have the clusterrole bound to the operator's service account.  

Consider rephrasing it this or even remove this if entirely and try to actually get the cluster role. Perhaps in the absence of necessary permissions we can recognize that and only then fail with some informative error?